src/gestures/use-pan-gesture.ts

```diff
@@ -397,150 +394,118 @@ export function usePanGesture(
-        },
-        [onPointerUp, onPointerMove]
-    )
+        const { point } = info
+        const { timestamp } = getFrameData()
+        session.current.pointHistory.push({ ...point, timestamp })
+
+        if (!panStarted) {
+            onPanStart && onPanStart(lastMoveEvent.current, info)
+            session.current.startEvent = lastMoveEvent.current
+        }
+
+        onPan && onPan(lastMoveEvent.current, info)
+    }
+
+    function onPointerMove(
+        event: MouseEvent | TouchEvent | PointerEvent,
+        info: EventInfo
+    ) {
+        lastMoveEvent.current = event
+        lastMoveEventInfo.current = transformPoint(info)
+
+        // Because Safari doesn't trigger mouseup events when it's above a `<select>`
+        if (isMouseEvent(event) && event.buttons === 0) {
+            onPointerUp(event, info)
+            return
+        }
+
+        // Throttle mouse move event to once per frame
+        sync.update(updatePoint, true)
+    }
+
+    function onPointerUp(
+        event: MouseEvent | TouchEvent | PointerEvent,
+        info: EventInfo
+    ) {
+        cancelPan()
+
+        if (!session.current) {
+            warning(false, "onPointerUp fired without pointer session")
+            return
+        }
+
+        onPanEnd && onPanEnd(event, getPanInfo(transformPoint(info)))
+
+        session.current = null
+    }
+
+    function onPointerDown(
+        event: MouseEvent | TouchEvent | PointerEvent,
+        info: EventInfo
+    ) {
+        // If we have more than one touch, don't start detecting this gesture
+        if (isTouchEvent(event) && event.touches.length > 1) return
+
+        const initialInfo = transformPoint(info)
+        const { point } = initialInfo
+
+        const { timestamp } = getFrameData()
+        session.current = {
+            target: event.target,
+            pointHistory: [{ ...point, timestamp }],
+        }
 
-    // On unmount, cancel pan gesture
-    useEffect(() => {
-        return () => {
-            stopPointerMove()
-            stopPointerUp()
-            cancelSync.update(updatePoint)
+        onPanSessionStart && onPanSessionStart(event, getPanInfo(initialInfo))
+
+        removePointerEvents()
+        const removeOnPointerMove = addPointerEvent(
+            window,
+            "pointermove",
+            onPointerMove
+        )
+        const removeOnPointerUp = addPointerEvent(
+            window,
+            "pointerup",
+            onPointerUp
+        )
+
+        pointerEventSubscription.current = () => {
+            removeOnPointerMove && removeOnPointerMove()
+            removeOnPointerUp && removeOnPointerUp()
         }
-    }, [])
-    let handlers: Partial<{ onPointerDown: EventHandler }> = { onPointerDown }
-    if (!(onPan || onPanStart || onPanEnd || onPanSessionStart)) {
-        handlers = {}
     }
 
-    return useConditionalPointerEvents(handlers, ref)
+    usePointerEvent(ref, "pointerdown", hasPanEvents && onPointerDown)
```

Comment:
The third argument looks like a boolean to me, but it is `undefined | function` this tripped me up, is there a clearer notation for this?

Reply:
Maybe a ternary operator to make it explicit?

`hasPanEvents ? onPointerDown : undefined`